### DISCUSSION TITLE
Date format in edge

### DISCUSSION THREAD

**Discussion by dmeinke-etc** [adonisjs/core#3480] [Help] [answered]
Is there a simple way to format a date in edge?
Currently it prints this out 
Fri Feb 09 1973 00:00:00 GMT+0000 (Coordinated Universal Time)
I need it to print 2/9/1973 

I was hoping 
`{{ (date(t.target_date, mm/dd/YYYY) }}` 

would be all I'd need but I cant find anything related to that?

If I have to I'll do database.rawQuery('select DATE_FORMAT(t.target_date,.......) as t_date')
But I dont feel that should be necessary

**Comment by Julien-R44** [comment, marked as answer]
If you are using a Luxon DateTime you could do like that : 

```
{{ myDate.toFormat('d/M/yyyy') }}
```

More information on Luxon DateTime.toFormat : https://moment.github.io/luxon/api-docs/index.html#datetimetoformat

**Reply by dmeinke-etc** [reply]
That looks like what I want but I don't see any instructions on installing luxon. I would assume it is with the new release? 

I tried 
`{{ t.targeet_date.toFormat('M/d/yyyy') }}`

And I get toFormat is not a function

**Reply by Julien-R44** [reply]
Yes, it's obviously not a Luxon DateTime
What version of Adonis are you using?
Please show your Controller

**Reply by dmeinke-etc** [reply]
Version 5 

`import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
import fundPipeline from "App/Models/FundPipeline";
import Fund from "App/Models/Fund";
import Database from "@ioc:Adonis/Lucid/Database";

export default class FundsPipelinesController {
  public async index({view}: HttpContextContract) {
    const funds = await Fund.query()
      .preload('client')
      .preload('custodian')
      .preload('distributor')

   const pipelinefunds = fundPipeline.all()
 return view.render('maintenance/fundpipeline', {funds, pipelinefunds})
}
`
Several date fields in that pipelinefunds query

**Reply by Julien-R44** [reply]
Yeah it should be good. 
I think you forgot to define your columns as dateTime in your models. 
You should have something like this in your template that contains the `target_date`:
```
  @column.dateTime()
  public targetDate: DateTime
```
After defining this you can use the `.toFormat` function in your edge template.

Check this : https://docs.adonisjs.com/guides/models/introduction#defining-columns

**Reply by dmeinke-etc** [reply]
THANKS! 
I had 
 @column.()
  public targetDate: DateTime

**Reply by FeivelRiodan** [reply]
i may be late but can anyone explain all these things to a total beginner?
